core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -242,6 +243,22 @@ private[deploy] class Worker(
           System.exit(1)
         }
     }
+    resources.keys.foreach { rName =>
+      resourcesUsed(rName) = MutableResourceInfo(rName, new HashSet[String])
+    }
+  }
+
+  private def updateResourcesUsed(deltaInfo: Map[String, ResourceInformation], add: Boolean)
```

Comment:
I'd say just create 2 functions for this, when looking at how this gets called its not nearly as clear as just saying addResourcesUsed, removeResourceUsed, user has to figure out what other parameter is.
also shrinks the code as you simply need 2 lines:
deltaInfo.foreach { case (rName, rInfo) =>
      resourcesUsed(rName) += resourcesUsed(rName) + rInfo
    }

Reply:
good point!